src/components/NanoContract/NanoContractDetailsHeader.js

```diff
@@ -64,12 +67,28 @@ export const NanoContractDetailsHeader = ({ nc, address, onAddressChange }) => {
     setShowUnregisterNanoContractModal(!showUnregisterNanoContractModal);
   };
 
+  const handleEditAddress = (item) => {
+    setSelectedItem(item);
+    setModalStep('edit');
+  };
+
+  const onDismissEditAddressModal = () => {
+    setSelectedItem({ address });
```

Comment:
question(non-blocking): This reverts to the original address, right? Maybe add a comment here reinforcing this?

All other address handlers receive their properties from the parameter, which may confuse future maintainers.

Reply:
Yes, this behavior is intentional. When the user dismisses the edit modal (by clicking "Go back"), we want to reset the selectedItem back to the original address prop so that if they open the edit flow again, it starts from the current address rather than whatever address they had previously selected but didn't confirm.